bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,483 @@
+            channelPostsCounter++;
+        }
+
+        // Fill channelPostsExt with each item value
+        int channelPostsCounterExt = 0;
+        for (SecondGenerationDxsEntries dxsentriesExt : dxsEntriesContainerExt.dxsEntries) {
+            channelPostsExt[channelPostsCounterExt] = dxsentriesExt.getName();
+            channelPostsCounterExt++;
+        }
+
+        // Fill channelPostsExtExt with each item value
+        int channelPostsCounterExtExt = 0;
+        for (SecondGenerationDxsEntries dxsentriesExtExt : dxsEntriesContainerExtExt.dxsEntries) {
+            channelPostsExtExt[channelPostsCounterExtExt] = dxsentriesExtExt.getName();
+            channelPostsCounterExtExt++;
+        }
+
+        // Create and update actual values for each channelPost
+        int channelValuesCounter = 0;
+        for (SecondGenerationChannelConfig cConfig : channelConfigs) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPosts[channelValuesCounter], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounter++;
+            }
+        }
+
+        // Create and update actual values for each channelPostExt
+        int channelValuesCounterExt = 0;
+        for (SecondGenerationChannelConfig cConfig : channelConfigsExt) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPostsExt[channelValuesCounterExt], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounterExt++;
+            }
+        }
+
+        // Create and update actual values for each channelPostExtExt
+        int channelValuesCounterExtExt = 0;
+        for (SecondGenerationChannelConfig cConfig : channelConfigsExtExt) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPostsExtExt[channelValuesCounterExtExt], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounterExtExt++;
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
+    public static String callURL(String myURL) {
+        StringBuilder sb = new StringBuilder();
+        URLConnection urlConn = null;
+        InputStreamReader in = null;
+        try {
+            URL url = new URL(myURL);
+            urlConn = url.openConnection();
+            if (urlConn != null) {
+                urlConn.setReadTimeout(60 * 1000);
+            }
+            if (urlConn != null && urlConn.getInputStream() != null) {
+                in = new InputStreamReader(urlConn.getInputStream(), Charset.defaultCharset());
```

Comment:
Please wrap your InputStream/Reader in a try-with-resources to make sure that it closes properly.